File: python_scripts/dec_to_bin.py
num = int(input("Enter a integer number: "))

bin_str = ""

if num >= 0:
    if num == 0:
        bin_str += '0'
    else:
        while(num != 0):
            bin_str += str(num & 1)
            num = num >> 1
    bin_str += '0'
else:
    while(~num != 0):
        bin_str += str(num & 1)
        num = num >> 1
    bin_str += '1'

print(bin_str[::-1])

File: c/dec_to_bin.c
#include <stdio.h>
#include <stdint.h>

void dec_to_bin(int num, uint8_t bin[32]){
	int i = 0;
	while((num != 0) && (i < 32)){
		if(num & 1){
			bin[i] = 1;
		}
		i++;
		num = num >> 1;
	}
}

int main(){
	int num;
	uint8_t bin[32] = {0};
	printf("Enter a integer number: ");
	scanf("%d", &num);
	dec_to_bin(num, bin);

	for(int i =31; i >= 0; i--){
		printf("%d", bin[i]);
	}
	printf("\n");
	return 0;	
}
